studio/components/interfaces/Settings/Database/SSLConfiguration.tsx

```diff
@@ -32,7 +31,18 @@ const SSLConfiguration = () => {
   } = useSSLEnforcementQuery({
     projectRef: ref,
   })
-  const { mutateAsync: updateSSLEnforcement } = useSSLEnforcementUpdateMutation()
+  const { mutate: updateSSLEnforcement, isLoading: isSubmitting } = useSSLEnforcementUpdateMutation(
+    {
+      onError: (error) => {
+        setIsEnforced(isEnforced)
```

Comment:
this is cause we setEnforced(!isEnforced) when we click on the toggle (so the UI doesnt wait for the async request to complete, otherwise the toggle will feel weird)

but if it errors out, then it should revert back to its original behaviour - in this case we do setEnforced(isEnforced) because all react's call stack with its useState hook, the value of isEnforced will be the original value prior to setEnforced(!isEnforced)

but good catch here though i forgot to add a toast for success

Reply:
I'll be honest, this still makes 0 sense to me hahaha
Like in my mind `setIsEnforced(isEnforced)` should just be setting `isEnforced` to what it already is. And if this is relying on stale state from React (?) that feels very riskay...

Reply:
> to what it already is
oh yeah hmm good point - lemme look into this actually, yeah you're right

Reply:
okay just pushed once more!